src/main/java/uk/gov/ons/ctp/response/action/service/ActionService.java

```diff
@@ -173,25 +200,129 @@ public Action createAction(final Action action) {
+
+    // If creating actions for business respondents, create an action per respondent
+    // Else create single action
+    if (actionCase.getSampleUnitType().equals(SampleUnitDTO.SampleUnitType.B.toString())
+        && (actionType.getActionTypePK() == 5 || actionType.getActionTypePK() == 7)) {
 
-    actionRepo
-        .findPotentialActionsActiveDate(actionPlan.getActionPlanPK(), currentTime)
+      List<Association> enrolledAssociations = respondentsEnrolledOnCase(actionCase);
+      if (enrolledAssociations.isEmpty()) {
+        log.error(NO_RESPONDENTS_FOR_SURVEY);
+        throw new IllegalStateException(NO_RESPONDENTS_FOR_SURVEY);
+      }
+
+      enrolledAssociations.forEach(
+          association ->
+              createAction(
+                  actionCase, actionRule, actionType, UUID.fromString(association.getPartyId())));
+
+    } else {
+      createAction(actionCase, actionRule, actionType, actionCase.getPartyId());
+    }
+  }
+
+  private List<Association> respondentsEnrolledOnCase(ActionCase actionCase) {
+    PartyDTO businessParty =
+        partySvcClientService.getParty(actionCase.getSampleUnitType(), actionCase.getPartyId());
+    CollectionExerciseDTO collectionExercise =
+        collectionExerciseClientService.getCollectionExercise(actionCase.getCollectionExerciseId());
+    return associationsEnrolledForSurvey(businessParty, collectionExercise.getSurveyId());
+  }
+
+  private List<Association> associationsEnrolledForSurvey(PartyDTO party, String surveyId) {
+    return party
+        .getAssociations()
         .stream()
-        .map(pa -> Action.fromPotentialAction(pa, currentTime))
-        .forEach(a -> actionRepo.save(a));
-    actionRepo.flush();
+        .filter(association -> isAssociationEnabledForSurvey(association, surveyId))
+        .collect(Collectors.toList());
+  }
+
+  private boolean isAssociationEnabledForSurvey(Association association, String surveyId) {
+    return association
+        .getEnrolments()
+        .stream()
+        .anyMatch(
+            enrolment ->
+                enrolment.getSurveyId().equals(surveyId)
+                    && enrolment.getEnrolmentStatus().equalsIgnoreCase(ENABLED));
+  }
+
+  private void createAction(
+      ActionCase actionCase, ActionRule actionRule, ActionType actionType, UUID partyId) {
+    if (actionRepo.findOneByCaseIdAndActionRuleFKAndPartyId(
+            actionCase.getId(), actionRule.getActionRulePK(), partyId)
+        != null) {
+      log.with("caseId", actionCase.getId().toString())
+          .with("actionRuleId", actionRule.getId().toString())
+          .with("partyId", partyId.toString())
+          .debug("Action already exists");
+      return;
+    }
+
+    Action newAction = new Action();
+    newAction.setId(UUID.randomUUID());
+    newAction.setCreatedBy("SYSTEM");
+    newAction.setManuallyCreated(false);
+    newAction.setState(ActionState.SUBMITTED);
+    newAction.setCreatedDateTime(new Timestamp((new Date()).getTime()));
```

Comment:
is this timezone safe? You used a function called `nowUTC()` somewhere else in the code